azconfig/client/src/main/java/com/azure/azconfig/AzConfigClient.java

```diff
@@ -0,0 +1,607 @@
+        Flux<RestPagedResponse<Key>> p = listKeysSinglePageAsync(filter)
+                       .concatMap(page -> {
+                           String nextPageLink = page.nextLink();
+                           if (nextPageLink == null) {
+                               return Flux.just(page);
+                           }
+                           return Flux.just(page).concatWith(listKeysNextAsync(nextPageLink));
+                       });
+       return  receiver.apply(p);
+    }
+
+    /**
+     * Gets all Keys.
+     *
+     * @param nextPageLink The nextPageLink from the previous successful call to List operation.
+     * @return the observable to the Page&lt;Key&gt; object.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     */
+    private Flux<RestPagedResponse<Key>> listKeysNextAsync(@NonNull String nextPageLink) {
+        return listKeysNextSinglePageAsync(nextPageLink)
+                .concatMap(page -> {
+                    String nextPageLink1 = page.nextLink();
+                    if (nextPageLink1 == null) {
+                        return Flux.just(page);
+                    }
+                    return Flux.just(page).concatWith(listKeysNextAsync(nextPageLink1));
+                });
+    }
+
+    /**
+     * Gets all Keys.
+     *
+     * @return the Flux&lt;RestPagedResponse&lt;Key&gt;&gt; object if successful.
+     */
+    private Flux<RestPagedResponse<Key>> listKeysSinglePageAsync(KeyLabelFilter filter) {
+        Mono<RestResponse<Map<String, String>, PageImpl<Key>>> result;
+        if (filter != null) {
+            result = service.listKeys(baseUri.toString(), filter.name(), filter.fields(), filter.acceptDatetime(), filter.range());
+        } else {
+            result = service.listKeys(baseUri.toString(), null, null, null, null);
+        }
+        return result.flatMapMany(p -> Flux.just(new RestPagedResponseImpl<>(p.body().items(), p.body().nextPageLink(), p.request(), p.headers(), p.statusCode())));
+    }
+
+    private Flux<RestPagedResponse<Key>> listKeysNextSinglePageAsync(String nextPageLink) {
+        if (nextPageLink == null) {
+            throw new IllegalArgumentException("Parameter nextPageLink is required and cannot be null.");
+        }
+        return service.listKeysNext(baseUri.toString(), nextPageLink)
+                       .flatMapMany(p -> Flux.just(new RestPagedResponseImpl<>(p.body().items(), p.body().nextPageLink(), p.request(), p.headers(), p.statusCode())));
+    }
+
+    /**
+     * Creates an pipeline to process the HTTP requests and Responses.
+     *
+     * @param credentials credentials the pipeline will use to authenticate the requests
+     * @return the pipeline
+     */
+    private static HttpPipeline createPipeline(AzConfigCredentials credentials, PipelineOptions pipelineOptions) {
+        if (pipelineOptions == null) {
+            throw new IllegalArgumentException("pipelineOptions cannot be null.");
+        }
```

Comment:
PipelineOptions is unused in this method, but cannot be null :) As I say elsewhere in my review, I believe this class should be removed entirely.